applications/neuroscan/backend/api/synapse/controllers/synapse.js

```diff
@@ -5,4 +5,33 @@
  * to customize this controller
  */
 
-module.exports = {};
+ const { sanitizeEntity } = require('strapi-utils');
+
+ module.exports = {
+   async find(ctx) {
+     let entities;
+     if (ctx.query._q) {
+       entities = await strapi.services.synapse.search(ctx.query);
+     } else {
+       entities = await strapi.services.synapse.find(ctx.query);
+     }
+
+     entities = entities.map(entity => {
+      const postNeuronPart = entity.postNeuron ? `-${entity.postNeuron.uid}` : '';
+      const neuronsPostPart = entity.neuronPost.length > 0 ? entity.neuronPost.reduce((r, n, i) => {
+        let s = (i != 0 ? ', ' : '');
+        let e = (i == entity.neuronPost.length - 1 ? ')' : '');
+        return `${r}${s}${n.uid}${e}`
+      }, ' (') : '';
+      return ({
+         ...entity,
+         uid: `pre-${entity.neuronPre.uid}-${entity.type}${postNeuronPart}${neuronsPostPart}`,
```

Comment:
could we add the neuronsPostPart between parenthesis as discussed at the call yesterday? Thx

Reply:
@ddelpiano it already is... see line 25 'opening parentheses' and line 23 'closing'